qubesmanager/backup.py

```diff
@@ -21,379 +20,299 @@
-        self.connect(self.select_vms_widget, SIGNAL("items_removed(PyQt_PyObject)"), self.vms_removed)
-        self.connect(self.select_vms_widget, SIGNAL("items_added(PyQt_PyObject)"), self.vms_added)
-        self.refresh_button.clicked.connect(self.check_running)
-        self.shutdown_running_vms_button.clicked.connect(self.shutdown_all_running_selected)
-        self.connect(self, SIGNAL("backup_progress(int)"), self.progress_bar.setValue)
-        self.dir_line_edit.connect(self.dir_line_edit, SIGNAL("textChanged(QString)"), self.backup_location_changed)
+        self.connect(self, QtCore.SIGNAL("currentIdChanged(int)"),
+                     self.current_page_changed)
+        self.connect(self.select_vms_widget,
+                     QtCore.SIGNAL("items_removed(PyQt_PyObject)"),
+                     self.vms_removed)
+        self.connect(self.select_vms_widget,
+                     QtCore.SIGNAL("items_added(PyQt_PyObject)"),
+                     self.vms_added)
+        self.dir_line_edit.connect(self.dir_line_edit,
+                                   QtCore.SIGNAL("textChanged(QString)"),
+                                   self.backup_location_changed)
 
         self.select_vms_page.isComplete = self.has_selected_vms
         self.select_dir_page.isComplete = self.has_selected_dir_and_pass
-        #FIXME
-        #this causes to run isComplete() twice, I don't know why
+        # FIXME
+        # this causes to run isComplete() twice, I don't know why
         self.select_vms_page.connect(
                 self.select_vms_widget,
-                SIGNAL("selected_changed()"),
-                SIGNAL("completeChanged()"))
+                QtCore.SIGNAL("selected_changed()"),
+                QtCore.SIGNAL("completeChanged()"))
         self.passphrase_line_edit.connect(
                 self.passphrase_line_edit,
-                SIGNAL("textChanged(QString)"),
+                QtCore.SIGNAL("textChanged(QString)"),
                 self.backup_location_changed)
         self.passphrase_line_edit_verify.connect(
                 self.passphrase_line_edit_verify,
-                SIGNAL("textChanged(QString)"),
+                QtCore.SIGNAL("textChanged(QString)"),
                 self.backup_location_changed)
 
         self.total_size = 0
-        self.__fill_vms_list__()
 
-        fill_appvms_list(self)
-        self.load_settings()
+        self.target_vm_list, self.target_vm_idx = utils.prepare_vm_choice(
+            self.appvm_combobox,
+            self.qvm_collection,
+            None,
+            self.qvm_collection.domains['dom0'],
+            (lambda vm: vm.klass != 'TemplateVM' and vm.is_running()),
+            allow_internal=False,
+            allow_default=False,
+            allow_none=False
+        )
+
+        selected = self.load_settings()
+        self.__fill_vms_list__(selected)
 
     def load_settings(self):
```

Comment:
This function could use some docstring, explaining that it does (especially that it first try to load backup profile, then application specific settings).